example/lib/peripheral_list/components/peripheral_row_view.dart

```diff
@@ -15,68 +15,54 @@ class PeripheralRowView extends StatelessWidget {
   Widget build(BuildContext context) {
     final navigatorBloc = BlocProvider.of<NavigationBloc>(context);
 
-    return Padding(
-      padding: const EdgeInsets.fromLTRB(0, 8.0, 0, 8.0),
-      child: ClipRRect(
-        borderRadius: BorderRadius.circular(8.0),
-        child: Container(
-          color: CustomColors.rowCardBackground,
-          child: ListTile(
-            leading: _buildListTileLeading(),
-            title: Text(_peripheral.name),
-            subtitle: Text(
-              _peripheral.id,
-              maxLines: 1,
-            ),
-            trailing: _buildListTileTrailing(),
-            dense: true,
-            onTap: () => _onRowTap(navigatorBloc),
-          ),
-        ),
-      ),
+    return PropertyRow(
+      title: BlePeripheralCategoryStringifier.name(_peripheral.category),
+      titleIcon: Icons.bluetooth,
+      titleColor: Theme.of(context).primaryColor,
+      value: _peripheral.name,
+      titleAccessory: _buildTitleAccessory(),
+      valueAccessory: _buildValueAccessory(),
+      onTap: () => _onRowTap(navigatorBloc),
     );
   }
 
   void _onRowTap(NavigationBloc navigatorBloc) {
     navigatorBloc.add(NavigateToPeripheralDetails(peripheral: _peripheral));
   }
 
-  Widget _buildListTileLeading() {
-    return CircleAvatar(
-      child: Icon(Icons.bluetooth),
+  Widget _buildTitleAccessory() {
+    return Icon(
+      Icons.chevron_right,
+      color: Colors.grey,
     );
   }
 
-  Widget _buildListTileTrailing() {
+  Widget _buildValueAccessory() {
     return Row(
+      crossAxisAlignment: CrossAxisAlignment.end,
       children: <Widget>[
-        Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            Icon(
-              Icons.signal_cellular_4_bar,
-              color: _colorForRssi(_peripheral.rssi),
-            ),
-            SizedBox(height: 4.0),
-            Text(
-              '${_peripheral.rssi ?? '-'} dbm',
-              style: TextStyle(fontSize: 12),
-            ),
-          ],
+        Text(
+          '${_peripheral.rssi ?? '-'} dbm',
+          style: CustomTextStyle.cardValueAccessory
+              .copyWith(color: _colorForRssi(_peripheral.rssi)),
+        ),
+        Padding(
+          padding: const EdgeInsets.only(left: 4.0),
+          child: Icon(
+            Icons.settings_input_antenna,
+            color: _colorForRssi(_peripheral.rssi),
+          ),
         ),
-        // Apply design guidelines properly for both platforms:
-        // iOS - list rows with disclosure indicator
-        // Android - list rows without disclosure indicator
-        if (Platform.isIOS)
-          Icon(Icons.chevron_right),
       ],
       mainAxisSize: MainAxisSize.min,
     );
   }
 
   Color _colorForRssi(int rssi) {
```

Comment:
Decision about parsing RSSI levels to some set of values should probably be extracted. 
```
SignalLevel parseRssi(int rsss);

enum SignalLevel { high, medium, low }
```

Reply:
Suggestion applied.